src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java

```diff
@@ -282,6 +283,17 @@ public void midEventAmendHearing(CaseData caseData, List<String> errors) {
         }
     }
 
+    public void setScotlandAllocatedOffice(String caseTypeId, CaseData caseData) {
+        if (!SCOTLAND_CASE_TYPE_ID.equals(caseTypeId)) {
+            return;
+        }
+
+        if (caseData.getManagingOffice() == null || UNASSIGNED_OFFICE.equals(caseData.getManagingOffice())) {
+            caseData.setManagingOffice(TribunalOffice.GLASGOW.getOfficeName());
```

Comment:
not sure on this. If a case comes into Scotland unassigned, this immediately changes the office to Glasgow and removes the logic of having an unassigned office unless I'm reading this wrong

Reply:
if (caseData.getManagingOffice() == null || !UNASSIGNED_OFFICE.equals(caseData.getManagingOffice())) {
            caseData.setAllocatedOffice(TribunalOffice.GLASGOW.getOfficeName());
        }
        
If the managing office is not unassigned, then assign it to Glasgow by default. If managingOffice = Unassigned, it should not set the allocated office

Reply:
I see. So adding allocated office is required only on the assign case event?

Reply:
If I have misunderstood the brief, a quick call may be .